### ISSUE TITLE
React native implementation of OSMD

### ISSUE THREAD

**Issue by nitzanrh** [opensheetmusicdisplay/opensheetmusicdisplay#891]
Hi guys,

I'm looking to implement OSMD in a react-native app (iOS an Android).
I looked into issues #491 #432 #670 and it seems to me there are basically 3 ways to approach this:

1. Using web-view inside react-native app: seems to me the 'easiest' way to go about it, I'm just wondering about the performance of such app (rendering time, animations etc).
2. Using PNG/SVG files: again seems reasonably easy to do, however I expect some problems with using functions such as GraphicSheet.calculateXPositionFromTimestamp (I've implemented my own cursor for playback and visual purposes), as well as some pinch-zoom functions.
3. Implementing OSMD directly into react-native: seems the 'best' approach and apparently it has been done before https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/issues/432#issuecomment-455258457 however I don't really know how to begin approaching this, or how much time consuming this is going to be ^^

So basically my REAL questions would be:
1. Is there a strong argument against going with WebView? 
2. How difficult would it be to implement OSMD directly into react-native?

I understand that these aren't technically OSMD related questions and that mobile support isn't a top priority for OSMD, however I would appreciate any help/guidelines on the matter :)

Cheers!
Nitzan

**Comment by sschmidTU** [comment]
Hi, i'm not a react specialist, so i'm not sure how much i can help here.
Have you taken a look at our React usage example (not react-native)?
https://github.com/opensheetmusicdisplay/react-opensheetmusicdisplay
Now, I understand that react-native is different, and https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/issues/432#issuecomment-455258457 suggests that it is a lot more effort for react-native, but maybe the usage example helps.

As for using WebView, i think that should be fine as long as you don't re-render often or render large scores.
Rendering scores that have many (>3) pages can take quite some time, but for smaller scores performance is usually fine.

**Comment by nitzanrh** [comment]
Hi @sschmidTU ,

Thanks for your comment and sorry for my late reply :)
I ended up using the React version and build my interface with React rather than React Native. So far it works flawlessly and the implementation was super easy. If at some point mobile support will become a priority it would be awesome to have a react-native branch.

Cheers!
